pkg/registry/core/service/strategy.go

```diff
@@ -158,8 +147,9 @@ func (svcStrategy) Export(ctx context.Context, obj runtime.Object, exact bool) e
 	if exact {
 		return nil
 	}
-	if t.Spec.ClusterIP != api.ClusterIPNone {
-		t.Spec.ClusterIP = ""
+	//set ClusterIPs as nil - if ClusterIPs[0] != None
+	if len(t.Spec.ClusterIPs) > 0 && t.Spec.ClusterIPs[0] != api.ClusterIPNone {
```

Comment:
I don't understand why?

Reply:
Honestly me neither. I modified it as we plural-ized the field. But i am not even sure which code path calls `Export(...)`